Sources/NIOWebSocket/WebSocketErrorCodes.swift

```diff
@@ -0,0 +1,214 @@
+    case unacceptableData
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received data within a message that was not
+    /// consistent with the type of the message (e.g. non-UTF-8
+    /// data within a text message).
+    /// Corresponds to code 1007.
+    case dataInconsistentWithMessage
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received a message that violates its policy.  This
+    /// is a generic status code that can be returned when there is no
+    /// other more suitable status code (e.g. 1003 or 1009) or if there
+    /// is a need to hide specific details about the policy.
+    /// Corresponds to code 1008.
+    case policyViolation
+
+    /// Indicates that an endpoint is terminating the connection
+    /// because it has received a message that is too big for it to
+    /// process.
+    /// Corresponds to code 1009.
+    case messageTooLarge
+
+    /// Indicates that an endpoint (client) is terminating the
+    /// connection because it has expected the server to negotiate one or
+    /// more extension, but the server didn't return them in the response
+    /// message of the WebSocket handshake.  The list of extensions that
+    /// are needed should appear in the `reason` part of the Close frame.
+    /// Note that this status code is not used by the server, because it
+    /// can fail the WebSocket handshake instead.
+    /// Corresponds to code 1010.
+    case missingExtension
+
+    /// Indicates that a server is terminating the connection because
+    /// it encountered an unexpected condition that prevented it from
+    /// fulfilling the request.
+    /// Corresponds to code 1011.
+    case unexpectedServerError
+
+    /// We don't have a better name for this error code.
+    case unknown(UInt16)
+
+    /// Create an error code from a raw 16-bit integer.
+    ///
+    /// - parameters:
+    ///     integer: The integer form of the status code.
+    public init(_ integer: UInt16) {
+        switch integer {
+        case 1000:
+            self = .normalClosure
+        case 1001:
+            self = .goingAway
+        case 1002:
+            self = .protocolError
+        case 1003:
+            self = .unacceptableData
+        case 1007:
+            self = .dataInconsistentWithMessage
+        case 1008:
+            self = .policyViolation
+        case 1009:
+            self = .messageTooLarge
+        case 1010:
+            self = .missingExtension
+        case 1011:
+            self = .unexpectedServerError
+        default:
+            self = .unknown(integer)
+        }
+    }
+
+    /// Create an error code from an integer.
+    ///
+    /// Will trap if the error code is not in the valid range.
+    ///
+    /// - parameters:
+    ///     - integer: The integer form of the status code.
+    public init(_ integer: Int) {
+        self.init(UInt16(integer))
+    }
+}
+
+public extension WebSocketErrorCode {
```

Comment:
given that you put the `read` and `get` extensions on `ByteBuffer`, why is the `write` on `WebSocketErrorCode`?

Reply:
That's the pattern used elsewhere in the codebase, e.g. in the `HTTP1` module.

Reply:
but is that a good pattern? I think it'd be better to have them on `ByteBuffer` as that's where the core `ByteBuffer` puts things. We could've put the extensions for ints, data, array, ... on those types too but decided to go with `ByteBuffer`. I think we should do that here too. The HTTP1 things should be deprecated & changed I think.